src/main/java/bio/terra/service/dataset/DatasetJsonConversion.java

```diff
@@ -182,14 +183,14 @@ public static TableModel tableModelFromTable(DatasetTable datasetTable) {
     public static Column columnModelToDatasetColumn(ColumnModel columnModel) {
         return new Column()
                 .name(columnModel.getName())
-                .type(columnModel.getDatatype())
+                .type(columnModel.getDatatype().toString())
```

Comment:
It looks like the only uses of `Column.type` are to extract the value for validation, such as in `CreateSnapshotFireStoreDataStep.doStep()`, and as a SQL param, such as in `DatasetTableDao.createColumns()`. Given that, it seems like it would be better to change the type of `Column.type` to `TableDatatypes` instead of converting it to `String` here. That would let us change `translateType` below back to a `switch` statement.

Reply:
Oh nice, I'm glad to see there's more opportunity to refactor! I just made this change naively so it would compile for testing